src/tools/rbd_mirror/image_replayer/BootstrapRequest.h

```diff
@@ -52,14 +52,15 @@ class BootstrapRequest : public BaseRequest {
         Journaler *journaler,
         MirrorPeerClientMeta *client_meta,
         Context *on_finish,
-        ProgressContext *progress_ctx = nullptr) {
+        ProgressContext *progress_ctx = nullptr,
+        bool *do_resync = nullptr) {
```

Comment:
Minor: why default this to null? Just force it to always be passed in and then you can remove all the null checks within the state machine. You'd want to update the unit test case as well anyway.